src/platform/silabs/rs911x/BLEManagerImpl.cpp

```diff
@@ -200,6 +200,71 @@ void BLEManagerImpl::sl_ble_event_handling_task(void * args)
+    SilabsBleWrapper::rsi_ble_add_char_serv_att(
+        new_serv_resp.serv_handler, new_serv_resp.start_handle + RSI_BLE_CHARACTERISTIC_RX_ATTRIBUTE_HANDLE_LOCATION,
+        RSI_BLE_ATT_PROPERTY_WRITE | RSI_BLE_ATT_PROPERTY_READ, // Set read, write, write without response
+        new_serv_resp.start_handle + RSI_BLE_CHARACTERISTIC_RX_VALUE_HANDLE_LOCATION, custom_characteristic_RX);
+
+    // Adding characteristic value attribute to the service
+    SilabsBleWrapper::rsi_ble_add_char_val_att(
+        new_serv_resp.serv_handler, new_serv_resp.start_handle + RSI_BLE_CHARACTERISTIC_RX_VALUE_HANDLE_LOCATION,
+        custom_characteristic_RX,
+        RSI_BLE_ATT_PROPERTY_WRITE | RSI_BLE_ATT_PROPERTY_READ, // Set read, write, write without response
+        data, sizeof(data), ATT_REC_IN_HOST);
+
+    static const uuid_t custom_characteristic_TX = {
+        .size     = RSI_BLE_CUSTOM_CHARACTERISTIC_TX_SIZE,
+        .reserved = { RSI_BLE_CUSTOM_CHARACTERISTIC_TX_RESERVED },
+        .val      = { .val128 = { .data1 = { RSI_BLE_CUSTOM_CHARACTERISTIC_TX_VALUE_128_DATA_1 },
+                                  .data2 = { RSI_BLE_CUSTOM_CHARACTERISTIC_TX_VALUE_128_DATA_2 },
+                                  .data3 = { RSI_BLE_CUSTOM_CHARACTERISTIC_TX_VALUE_128_DATA_3 },
+                                  .data4 = { RSI_BLE_CUSTOM_CHARACTERISTIC_TX_VALUE_128_DATA_4 } } }
+    };
+
+    // Adding custom characteristic declaration to the custom service
+    SilabsBleWrapper::rsi_ble_add_char_serv_att(
+        new_serv_resp.serv_handler, new_serv_resp.start_handle + RSI_BLE_CHARACTERISTIC_TX_ATTRIBUTE_HANDLE_LOCATION,
+        RSI_BLE_ATT_PROPERTY_WRITE_NO_RESPONSE | RSI_BLE_ATT_PROPERTY_WRITE | RSI_BLE_ATT_PROPERTY_READ |
+            RSI_BLE_ATT_PROPERTY_NOTIFY | RSI_BLE_ATT_PROPERTY_INDICATE, // Set read, write, write without response
+        new_serv_resp.start_handle + RSI_BLE_CHARACTERISTIC_TX_MEASUREMENT_HANDLE_LOCATION, custom_characteristic_TX);
+
+    // Adding characteristic value attribute to the service
+    rsi_ble_measurement_hndl = new_serv_resp.start_handle + RSI_BLE_CHARACTERISTIC_TX_MEASUREMENT_HANDLE_LOCATION;
+
+    // Adding characteristic value attribute to the service
+    rsi_ble_gatt_server_client_config_hndl =
+        new_serv_resp.start_handle + RSI_BLE_CHARACTERISTIC_TX_GATT_SERVER_CLIENT_HANDLE_LOCATION;
+
+    SilabsBleWrapper::rsi_ble_add_char_val_att(new_serv_resp.serv_handler, rsi_ble_measurement_hndl, custom_characteristic_TX,
+                                               RSI_BLE_ATT_PROPERTY_WRITE_NO_RESPONSE | RSI_BLE_ATT_PROPERTY_WRITE |
+                                                   RSI_BLE_ATT_PROPERTY_READ | RSI_BLE_ATT_PROPERTY_NOTIFY |
+                                                   RSI_BLE_ATT_PROPERTY_INDICATE, // Set read, write, write without response
+                                               data, sizeof(data), ATT_REC_MAINTAIN_IN_HOST);
+
+    memset(&data, 0, sizeof(data));
```

Comment:
I'm not sure i see the usefullness of this memset. The data never seems to change between function executions.
If it not equivalent of a constant (which it seems like), the data is cleared after the stack leaves the function.

Reply:
removed